### PR TITLE
WAF AWS managed rule set module

### PR DESCRIPTION
**Pull request by TomJKing**

### CODE REVIEW

File: wafv2_aws_managed_rules/variables.tf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  default     = []
+}
+
+variable "arn_associations" {
+  description = "List of AWS resource ARNs for WAF rule association"
+}
+
+variable "aws_managed_rules" {
+  description = "List of AWS managed rules to be applied"
+  type        = list(any)
```
Comment (MancunianSam) [inline wafv2_aws_managed_rules/variables.tf:13]: You can get it to validate the type by being more specific with the type. Something like
```
object({
    name = string
    priority = number
    managed_rule_group_statement_name = string
    managed_rule_group_statement_vendor_name = string
    metric_name = string
  })
```
That way, terraform will complain if you don't put the right fields in. You can make some of them optional by doing `metric_name = optional(string, "")` if they're not all needed.

**Review by MancunianSam — COMMENTED** [review]
Looks good to me. Can you put a couple of lines in the main README file about what the module does and what the defaults do please.

File: README.md [hunk 2]
```diff
@@ -62,6 +62,18 @@ Creates an SQS queue.
...
 
+## WAFv2 AWS Managed Rules
+Creates a set of AWS managed WAF rules to be associated with the given AWS resources.
+
+Provides a default set of AWS managed rules. The default rules are "free" and non-language and non-OS specific:
+* `AWSManagedRulesAmazonIpReputationList`
+* `AWSManagedRulesCommonRuleSet`
+* `AWSManagedRulesCommonRuleSet`
+
+Calling clients are can define whatever set of AWS managed rules are needed and are not restricted to the default rules.
```
Comment (MancunianSam) [inline README.md:73]: I'm not sure that `Calling clients are can define` is quite right.

Comment (TomJKing) [reply README.md:73]: @MancunianSam How about something like this: `Calling modules can chose whatever set of AWS managed rules are most appropriate for the particular service and are not restricted to the default rules.`

Comment (MancunianSam) [reply README.md:73]: I'm not the best to ask about documentation but that seems sensible. I'd use `choose` instead of `chose` though.

**Review by MancunianSam — APPROVED** [review]

**Pull request merged**